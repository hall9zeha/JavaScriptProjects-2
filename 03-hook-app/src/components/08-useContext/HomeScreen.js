import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const HomeScreen = () => {
    //Buscará la instancia de UserContext en el árbol de componentes
    //ya que necesitamos acceder lo que hayamos compartido a través de ella
    const {user} = useContext(UserContext)
    console.log(user);
  return (
    <div>
        <h1>HomeScreen</h1>
        <pre>
          {JSON.stringify(user,null,3)}
        </pre>
        <hr/>
    </div>
  )
}
