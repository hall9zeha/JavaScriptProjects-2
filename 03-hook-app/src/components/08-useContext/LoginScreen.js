import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {
    const {user, setUser} = useContext(UserContext);

    const actionLogin = ()=>{
        const {id,name,email} = user;
        setUser({
            id,
            name,
            email,
            sigIn:true
        })
    }

  return (
    <div>
        <h1>LoginScreen</h1>
        <hr/>
        <button onClick={actionLogin} className='btn btn-primary '>Login</button>
    </div>
  )
}
