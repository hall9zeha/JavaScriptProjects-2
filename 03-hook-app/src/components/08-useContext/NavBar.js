import React from 'react'
import { Link, NavLink } from 'react-router-dom'

export const NavBar = () => {
  return (
        <nav className="navbar navbar-expand-sm  navbar-dark  bg-dark">
            <div className="container-fluid">
                <a className="navbar-brand" href="#">useContext</a>
              
                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                    {/* NavLink nos permite usar algunas clases de bootstrap */}
                    <NavLink exact activeClassName="active" className="nav-link nav-item " to='/'>Home</NavLink>
                    <NavLink exact activeClassName="active" className="nav-link nav-item" to='/about'>About</NavLink>
                    <NavLink exact activeClassName="active" className="nav-link nav-item" to='/login'>Login</NavLink>
                 
                </div>
                </div>
            </div>
        </nav>
   
  )
}
