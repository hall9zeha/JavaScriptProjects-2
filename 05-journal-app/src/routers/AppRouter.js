import React, { useEffect, useState } from 'react'
import {firebase} from '../firebase/firebaseConfig'
/*
Ya que react router 5 es todavía muy utilizado este proyecto implementará dicha versión
para aprender el manejor de las rutas con la misma.
Al momento de realizar este proyecto la versión más reciente era 6.22
*/
import{
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
} from "react-router-dom";
import { PrivateRoute } from './PrivateRoute';
import { AuthRouter } from './AuthRouter';
import { JournalScreen } from '../components/journal/JournalScreen';
import { useDispatch } from 'react-redux';
import { login } from '../actions/auth';
import { PublicRoute } from './PublicRoute';
import { loadNotes } from '../helpers/loadNotes';
import { setNotes, startLoadingNotes } from '../actions/notes';


export const AppRouter = () => {

    const dispatch = useDispatch();
    const [cheking, setCheking] = useState(true);
    const [isLogged, setIsLogged] = useState(false);

    // Observador que mostrara los estados de autenticación de firebase cuando estos cambien (login or logout)
    // nos devolverá el objeto correspondiente (user credentials or null)
    useEffect(()=>{
        firebase.auth().onAuthStateChanged(async (user)=>{
            // Si el usuario no es nulo
            if(user?.uid){
                dispatch(login(user.uid,user.displayName));
                setIsLogged(true)
                dispatch(startLoadingNotes(user.uid));  

            }else{
                setIsLogged(false)
            }
            // Si deveulve datos de usuario correctamente la espera se detendrá 
            setCheking(false);   
        })
    }, [dispatch,setCheking])//[] Dependencias vacías para que se ejecute una sola vez si se vuelve a renderizar algún componente o
    // ya que dispatch y setChecking no van a cambiar tampoco se ejecutará nuevamente nustro observable, podemos incluirlos dentro [.]
    
    //Para mostrar las vistas principales esperamos a la comprobación de si estamos autenticados o no
    if(cheking){
        return (
            <h1>Espere...</h1>
        )
    }
  return (
    <Router>
        <div>
            <Switch>
                <PublicRoute
                    path="/auth"
                    component={AuthRouter}
                    isAuthenticated={isLogged}
                />
                <PrivateRoute
                    isAuthenticated={isLogged}
                    exact
                    path="/"
                    component={JournalScreen}
                />
                <Redirect to="/auth/login"/>
            </Switch>
        </div>
       
    </Router>
  )
}
