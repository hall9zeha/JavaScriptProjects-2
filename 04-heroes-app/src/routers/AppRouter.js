
import React from 'react'
import { BrowserRouter, Route, Routes } from 'react-router-dom'

import { LoginScreen } from '../components/login/LoginScreen'

import { DashboardRoutes } from './DashboardRoutes'

export const AppRouter = () => {
  return (
    <BrowserRouter>
        {/* Mostraremos la pantalla de login pero sin el navbar */}
        <Routes>
            <Route path='/login' element={<LoginScreen/>}/>
            <Route path='/*' element={<DashboardRoutes/>}/>
        </Routes>
    </BrowserRouter>
  )
}
