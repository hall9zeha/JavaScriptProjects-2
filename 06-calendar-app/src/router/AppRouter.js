import React from 'react'
import {BrowserRouter , Navigate, Route, Router, Routes} from 'react-router-dom'
import { LoginScreen } from '../auth/LoginScreen'

import { CalendarScreen } from '../calendar/CalendarScreen'

export const AppRouter = () => {
  return (
    <BrowserRouter>
        <Routes>
             <Route 
                Component={LoginScreen}
                exact path='/login'/>
            <Route
                Component={CalendarScreen}
                exact path='/calendar'/>
            <Route 
                path='*' element={<Navigate to = "/calendar"/>}
            />
           </Routes>
    </BrowserRouter>
  )
}
