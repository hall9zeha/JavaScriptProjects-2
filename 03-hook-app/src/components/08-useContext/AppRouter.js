import React from 'react'
import { BrowserRouter as  Router, Routes, Route } from 'react-router-dom'
import { AboutScreen } from './AboutScreen'
import { HomeScreen } from './HomeScreen'
import { LoginScreen } from './LoginScreen'
import { NavBar } from './NavBar'

export const AppRouter = () => {
  return (
    <Router>
        <div>
            <NavBar/>
            {/* El Routes de react-router es como un switch tradicional(Switch en react-router-dom está descontinuado), dependiendo de qué condición ejecutará tal o cual
            En este caso una dirección a un componente dentro de la ruta */}
            <Routes>
            {/* exact le dice a la ruta que debe buscar el path con todos sus caracteres y no por la primera coincidencia */}
                <Route exact path='/' Component={HomeScreen}/> 
                <Route exact path='/login' Component={LoginScreen}/>
                <Route exact path='/about' Component={AboutScreen}/>
                {/* Ponemos una ruta por defecto si no existiera ninguna de las anteriores */}
                <Route path='*' Component={HomeScreen}/> 
            </Routes>
        </div>
    </Router>
  )
}
