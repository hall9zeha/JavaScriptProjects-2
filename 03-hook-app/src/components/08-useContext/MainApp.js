import React, { useState } from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './UserContext'

export const MainApp = () => {
    const userProvides = {
        id:123,
        name:"Martha",
        email:"martha@mail.com",
        sigIn:false
    }

    const [userState, setUserState] = useState(userProvides);

  return (
    //Envolver nuestros componentes con UseProvider nos permitir√° compartir 
    //elementos u objetos con sus componentes hijos 

    //Ahora enviaremos un objeto que contenga las propiedades de nuestro useState
    <UserContext.Provider value={
        {user:userState,
         setUser:setUserState
        }}>
        <AppRouter/>
    </UserContext.Provider>
  )
}

