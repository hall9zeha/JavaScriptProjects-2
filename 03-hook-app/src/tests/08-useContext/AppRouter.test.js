import React from "react";
import renderer from 'react-test-renderer'
import { AppRouter } from "../../components/08-useContext/AppRouter";
import { UserContext } from "../../components/08-useContext/UserContext";

describe('Tests in <AppRouter/> component', () => { 
    const userMock={
        id:1,
        name:'Barry',
        email:'barry@mail.com',
        sigIn:true
    }


    test('should show successfully',()=>{
        const wrapper = renderer.create(
            <UserContext.Provider value={{user:userMock}}>
                <AppRouter/>
            </UserContext.Provider>
        ).toJSON()

        expect(wrapper).toMatchSnapshot();
    })
})