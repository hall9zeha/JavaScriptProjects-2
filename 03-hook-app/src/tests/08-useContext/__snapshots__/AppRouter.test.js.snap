// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <AppRouter/> component should show successfully 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm  navbar-dark  bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="#"
      >
        useContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            activeClassName="active"
            aria-current="page"
            className="nav-link nav-item  active"
            exact={true}
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            activeClassName="active"
            className="nav-link nav-item"
            exact={true}
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            activeClassName="active"
            className="nav-link nav-item"
            exact={true}
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div>
    <h1>
      HomeScreen
    </h1>
    <pre>
      {
   "id": 1,
   "name": "Barry",
   "email": "barry@mail.com",
   "sigIn": true
}
    </pre>
    <hr />
  </div>
</div>
`;
